ios/VydiaRNFileUploader.m

```diff
@@ -226,7 +226,8 @@ - (void)copyAssetToFile: (NSString *)assetUrl completionHandler: (void(^)(NSStri
 RCT_EXPORT_METHOD(cancelUpload: (NSString *)cancelUploadId resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject) {
     [_urlSession getTasksWithCompletionHandler:^(NSArray *dataTasks, NSArray *uploadTasks, NSArray *downloadTasks) {
         for (NSURLSessionTask *uploadTask in uploadTasks) {
-            if (uploadTask.taskDescription == cancelUploadId) {
+            if ([uploadTask.taskDescription isEqualToString:cancelUploadId]){
+                // == checks if references are equal, while isEqualToString checks the string value
```

Comment:
Is this `==` comparison also an issue? 

https://github.com/Vydia/react-native-background-upload/blob/ddbe25b405d27ad564ce4589b0adc87be797cf1f/ios/VydiaRNFileUploader.m#L307-L312

Reply:
It doesn't seem like an issue to me, it is entering the `if` condition correctly.